dbms/src/Common/Exception.cpp

```diff
@@ -24,7 +24,17 @@ namespace ErrorCodes
 void throwFromErrno(const std::string & s, int code, int e)
 {
 	char buf[128];
+#ifndef _GNU_SOURCE
+	if (!strerror_r(e, buf, sizeof(buf))) {
+		std::string tmp = std::to_string(code);
+		const char* code = tmp.c_str();
+		strcpy(buf, "Unknown error ");
+		strcpy(buf+13, code);
```

Comment:
Constant 13 is calculated wrong (need to be strlen("Unknown error "), actually 14).
But we already convinced that this is a bad solution.
I hope you will find strerror replacement for Mac :)